synapse/handlers/sync.py

```diff
@@ -1619,7 +1619,7 @@ async def _get_ignored_users(self, user_id: str) -> FrozenSet[str]:
         # TODO: Can we `SELECT ignored_user_id FROM ignored_users WHERE ignorer_user_id=?;` instead?
         ignored_account_data = (
             await self.store.get_global_account_data_by_type_for_user(
-                AccountDataTypes.IGNORED_USER_LIST, user_id=user_id
+                user_id=user_id, data_type=AccountDataTypes.IGNORED_USER_LIST
```

Comment:
How does this interact with the caching? It uses kwargs so I don't know how our caching handles that

Reply:
I think the cached functions just base it off name, so this should work OK. I would be fine with you making them into normal args if you'd prefer though.

Reply:
I dug in to the (slightly confusing) code a bit and it looks like you're right :). Good to know.